action_macro/src/lib.rs

```diff
@@ -55,23 +57,10 @@ pub fn action_impl(
     }
 
     let sync_run = if let Some(sync_type) = sync_type {
-        // In expression position, the type needs to be of the form Foo::<Bar>, not Foo<Bar>
-        let mut formal = sync_type.clone();
-        struct Visitor;
-        impl VisitMut for Visitor {
-            fn visit_path_segment_mut(&mut self, segment: &mut syn::PathSegment) {
-                if let PathArguments::AngleBracketed(args) = &mut segment.arguments {
-                    if args.colon2_token.is_none() {
-                        args.colon2_token = Some(Token![::](Span::call_site()));
-                    }
-                }
-            }
-        }
-        syn::visit_mut::visit_type_mut(&mut Visitor, &mut formal);
         quote! {
             /// Synchronously execute this action.
             pub fn run(self) -> Result<#sync_type> {
-                crate::sync::TOKIO_RUNTIME.block_on(std::future::IntoFuture::into_future(self)).map(#formal::new)
+                crate::sync::TOKIO_RUNTIME.block_on(std::future::IntoFuture::into_future(self)).map(<#sync_type>::new)
```

Comment:
Coincidentally, when reading the documentation for `quote` I discovered that this is a much easier way of handling the type-position vs expr-position issue for qualified types.